--- origsrc/celestia-1.6.1/src/celengine/frametree.h	2011-06-05 11:11:09.000000000 -0500
+++ src/celestia-1.6.1/src/celengine/frametree.h	2013-10-09 14:08:19.715412000 -0500
@@ -13,6 +13,7 @@
 #ifndef _CELENGINE_FRAMETREE_H_
 #define _CELENGINE_FRAMETREE_H_
 
+#include <cstddef>
 #include <vector>
 
 class Star;
--- origsrc/celestia-1.6.1/src/celmath/intersect.h	2011-06-05 11:11:15.000000000 -0500
+++ src/celestia-1.6.1/src/celmath/intersect.h	2013-10-09 14:06:37.323555500 -0500
@@ -12,6 +12,7 @@
 #ifndef _CELMATH_INTERSECT_H_
 #define _CELMATH_INTERSECT_H_
 
+#include "mathlib.h"
 #include "ray.h"
 #include "sphere.h"
 #include "ellipsoid.h"
